toc_config.py

```diff
@@ -370,6 +370,31 @@ def traverse_tocs(app, doc):
     names = []
     for toc in doc.traverse(addnodes.toctree):
         hidden = toc.get('hidden', False)
-        for _,docname in toc.get('entries', []):
-            names.append((docname,hidden))
-    return [(name,hidden,app.env.get_doctree(name)) for name,hidden in names]
+        for _, docname in toc.get('entries', []):
+            names.append((docname, hidden))
+    return [(name, hidden, app.env.get_doctree(name)) for name, hidden in names]
+
+
+# This function adds the language postfix to chapterlinks in multilingual
+# courses, if it hasn't been manually added. This is necessary, since Sphinx
+# requires, that the file needs to exists, in order to make the link. However,
+# this post fix is removed in the rewrite_outdir function in html_tools, since
+# a-plus uses chapter keys in the url:s, and the postfixes are removed from
+# chapter keys when the language indexes are merged.
+# The source argument is a list whose single element is the contents of the source file
+def add_lang_postfixes_to_links(app, docname, source):
+    # If the course use different format in links or for some other reason links
+    # need to stay untouched, add no_link_correction to conf.py or
+    # no-link-correction to the main index
+    if ("no_link_correction" not in app.env.metadata[app.config.master_doc]
+            and not app.config.no_link_correction):
+        postfix = docname[-3:]
+        if postfix[0] == '_':
+            # Links of the form :doc:`link text <path/file>` or `link text <path/file>`_
+            source[0] = re.sub(r"<([a-zA-Z0-9_/.]+/)?([a-zA-Z0-9_]+[^_]..)>",
```

Comment:
Now that I think of it, links of the form `link text <path/file>`_ shouldn't be touched at all, since they do not necessarily point to the course material, but in any page in general.